external civetweb (builtin_civetweb=off)  try_compile fails because of undefined SSL symbols

https://github.com/root-project/root/blob/58786cc3783e90f0afa7a5529f8d206c48a2453d/cmake/modules/SearchInstalledSoftware.cmake#L563

Why not use the `civetweb::civetweb` target here and get all the include directories, flags and transitive library requirements for free? I.e., at the moment this fails because of missing openssl symbols in the LCG setup when we build civetweb ourselves.


```
-- Looking for civetweb
CMake Error at cmake/modules/SearchInstalledSoftware.cmake:584 (message):
  Could not check for civetweb features: Change Dir:
  '/home/sailer/SPI/buildgcc15/projects/ROOT-HEAD/src/ROOT-HEAD-build/CMakeFiles/CMakeScratch/TryCompile-4PoQW8'




  Run Build Command(s):
  /cvmfs/sft.cern.ch/lcg/contrib/CMake/3.30.6/Linux-x86_64/bin/cmake -E env
  VERBOSE=1 /usr/bin/gmake -f Makefile cmTC_c1075/fast

  gmake[4]: Entering directory
  '/home/sailer/SPI/buildgcc15/projects/ROOT-HEAD/src/ROOT-HEAD-build/CMakeFiles/CMakeScratch/TryCompile-4PoQW8'


  /usr/bin/gmake -f CMakeFiles/cmTC_c1075.dir/build.make
  CMakeFiles/cmTC_c1075.dir/build

  gmake[5]: Entering directory
  '/home/sailer/SPI/buildgcc15/projects/ROOT-HEAD/src/ROOT-HEAD-build/CMakeFiles/CMakeScratch/TryCompile-4PoQW8'


  Building C object CMakeFiles/cmTC_c1075.dir/civetweb_check_features.c.o

  /cvmfs/sft.cern.ch/lcg/releases/gcc/15.2.0-35657/x86_64-el9/bin/gcc
  -I/home/sailer/SPI/install/dev4/civetweb/1.16/x86_64-el9-gcc15-opt/include
  -Wno-implicit-fallthrough -pipe -Wall -W -pthread -o
  CMakeFiles/cmTC_c1075.dir/civetweb_check_features.c.o -c
  /home/sailer/SPI/buildgcc15/projects/ROOT-HEAD/src/ROOT/HEAD/cmake/modules/civetweb_check_features.c


  Linking C executable cmTC_c1075

  /cvmfs/sft.cern.ch/lcg/contrib/CMake/3.30.6/Linux-x86_64/bin/cmake -E
  cmake_link_script CMakeFiles/cmTC_c1075.dir/link.txt --verbose=1

  /cvmfs/sft.cern.ch/lcg/releases/gcc/15.2.0-35657/x86_64-el9/bin/gcc
  -Wno-implicit-fallthrough -pipe -Wall -W -pthread -rdynamic
  CMakeFiles/cmTC_c1075.dir/civetweb_check_features.c.o -o cmTC_c1075
  /home/sailer/SPI/install/dev4/civetweb/1.16/x86_64-el9-gcc15-opt/lib64/libcivetweb.a
  in function `pull_inner':

  civetweb.c:(.text+0x1417): undefined reference to `SSL_pending'

  /cvmfs/sft.cern.ch/lcg/releases/binutils/2.40-acaab/x86_64-el9/bin/ld:
  civetweb.c:(.text+0x1432): undefined reference to `ERR_clear_error'

  /cvmfs/sft.cern.ch/lcg/releases/binutils/2.40-acaab/x86_64-el9/bin/ld:
  civetweb.c:(.text+0x1445): undefined reference to `SSL_read'

  /cvmfs/sft.cern.ch/lcg/releases/binutils/2.40-acaab/x86_64-el9/bin/ld:
  civetweb.c:(.text+0x14f1): undefined reference to `ERR_clear_error'

[snip]
```

Things are configuring when I replace this with the `civetweb::civetweb` target.


PS: added some of the errors, snipped too soon.

--- a/cmake/modules/SearchInstalledSoftware.cmake
+++ b/cmake/modules/SearchInstalledSoftware.cmake
@@ -546,28 +546,15 @@ if(http AND NOT builtin_civetweb)
     endif()
   endif()
   if(civetweb_FOUND)
-    get_target_property(CIVETWEB_IMPORTED_LOCATION civetweb::civetweb IMPORTED_LOCATION_NONE)
-    # Fall-back solution for some OS such as opensuse16 with different IMPORTED_LOCATION_xxxx config choice
-    if(NOT CIVETWEB_IMPORTED_LOCATION OR CIVETWEB_IMPORTED_LOCATION MATCHES "NOTFOUND")
-      get_target_property(CIVETWEB_IMPORTED_LOCATION civetweb::civetweb IMPORTED_LOCATION_RELEASE)
-      if(NOT CIVETWEB_IMPORTED_LOCATION OR CIVETWEB_IMPORTED_LOCATION MATCHES "NOTFOUND")
-        get_target_property(CIVETWEB_IMPORTED_LOCATION civetweb::civetweb IMPORTED_LOCATION_RELWITHDEBINFO)
-        if(NOT CIVETWEB_IMPORTED_LOCATION OR CIVETWEB_IMPORTED_LOCATION MATCHES "NOTFOUND")
-          get_target_property(CIVETWEB_IMPORTED_LOCATION civetweb::civetweb IMPORTED_LOCATION_DEBUG)
-        endif()
-      endif()
-    endif()
     try_compile(CIVETWEB_FEATURE_API
       SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules/civetweb_check_features.c"
-      CMAKE_FLAGS "-DINCLUDE_DIRECTORIES=${civetweb_INCLUDE_DIR}"
-      LINK_LIBRARIES ${CIVETWEB_IMPORTED_LOCATION}
+      LINK_LIBRARIES civetweb::civetweb
       OUTPUT_VARIABLE CIVETWEB_FEATURE_API_LOG
     )
     if (CIVETWEB_FEATURE_API)
       try_run(RUN_RESULT COMPILE_RESULT
         SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules/civetweb_check_features.c"
-        CMAKE_FLAGS "-DINCLUDE_DIRECTORIES=${civetweb_INCLUDE_DIR}"
-        LINK_LIBRARIES ${CIVETWEB_IMPORTED_LOCATION}
+        LINK_LIBRARIES civetweb::civetweb
         COMPILE_OUTPUT_VARIABLE BUILD_LOG
         RUN_OUTPUT_VARIABLE CIVETWEB_FEATURES
       )

